File: dashboard-service/src/main/java/fpt/aptech/dashboardservice/models/Room.java
package fpt.aptech.dashboardservice.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Time;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "rooms")
@Builder
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String roomName;
    private String slug;
    private int capacity;    //Sức chứa của phòng
    private int availableSeats;    //chỗ trống hiện tại của lớp học
    private String facilities;
    private boolean status;    //Trạng thái của room(VD: có thể sử dụng hoặc đang bảo trì )
    private LocalTime startTime;
    private LocalTime endTime;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "club_id", nullable = false)
    @JsonIgnore
    private Club club;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trainer_id", nullable = false)
    @JsonIgnore
    private Trainer trainer;

}

File: FLUTTER/lib/features/Home/data/Room.dart
import 'Trainer.dart';

class Room {
  int? id;
  String? roomname;
  String? slug;
  int? capacity;
  int? availableseats;
  String? facilities;
  bool? status;
  List<int>? starttimeList;
  List<int>? endtimeList;
  List<int>? createdatList;
  List<int>? updatedatList;
  late bool isBooked;
  Trainer? trainer;

  Room({
    this.id,
    this.roomname,
    this.slug,
    this.capacity,
    this.availableseats,
    this.facilities,
    this.status,
    this.starttimeList,
    this.endtimeList,
    this.createdatList,
    this.updatedatList,
    this.isBooked = false,
    this.trainer,
  });

  Map<String, dynamic> toJson() {
    final map = <String, dynamic>{};
    map["id"] = id;
    map["roomName"] = roomname;
    map["slug"] = slug;
    map["capacity"] = capacity;
    map["availableSeats"] = availableseats;
    map["facilities"] = facilities;
    map["status"] = status;
    map["startTime"] = starttimeList;
    map["endTime"] = endtimeList;
    map["createdAt"] = createdatList;
    map["updatedAt"] = updatedatList;
    map["isBooked"] = isBooked;
    map["trainer"] = trainer?.toJson();
    return map;
  }

  Room.fromJson(dynamic json){
    id = json["id"];
    roomname = json["roomName"];
    slug = json["slug"];
    capacity = json["capacity"];
    availableseats = json["availableSeats"];
    facilities = json["facilities"];
    status = json["status"];
    starttimeList =
    json["startTime"] != null ? json["startTime"].cast<int>() : [];
    endtimeList = json["endTime"] != null ? json["endTime"].cast<int>() : [];
    createdatList =
    json["createdAt"] != null ? json["createdAt"].cast<int>() : [];
    updatedatList =
    json["updatedAt"] != null ? json["updatedAt"].cast<int>() : [];
    isBooked = false;
    trainer = json["trainer"] != null ? Trainer.fromJson(json["trainer"]) : null;
  }
}